From d98c862eb155718d32b647e5f9f2070f92cdb284 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sat, 4 Jun 2022 11:05:04 -0400
Subject: cmake GCC build fix of bitmap_info.h.in

Remove wxT macro that should not be used in wx 3.0 or later.
Use std::string instead of wxString to avoid weird MinGW64 GCC bug.
Bug might be a resource issue.

---
 bitmaps_png/CMakeLists.txt       |  2 +-
 include/bitmaps/bitmap_info.h.in | 11 ++++++-----
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/bitmaps_png/CMakeLists.txt b/bitmaps_png/CMakeLists.txt
index 76db9bde16..2e1a037101 100644
--- a/bitmaps_png/CMakeLists.txt
+++ b/bitmaps_png/CMakeLists.txt
@@ -780,7 +780,7 @@ function( bitmap_dir pngWidth pngHeight bmapList )
             png2png( ${tmpFile} ${pngPath} )
 
             set( bitmapInfo
-            "    { BITMAPS::${bmn}, wxT( \"${pngFile}\" ), ${pngHeight}, wxT( \"${theme}\" ) },\n"
+            "    { BITMAPS::${bmn}, \"${pngFile}\", ${pngHeight}, \"${theme}\" },\n"
             )
 
             set( BITMAP_INFO_LIST "${BITMAP_INFO_LIST}${bitmapInfo}" )
diff --git a/include/bitmaps/bitmap_info.h.in b/include/bitmaps/bitmap_info.h.in
index 688472f919..d0b2ccdec8 100644
--- a/include/bitmaps/bitmap_info.h.in
+++ b/include/bitmaps/bitmap_info.h.in
@@ -26,16 +26,16 @@
 #define KICAD_BITMAP_INFO_H
 
 #include <vector>
+#include <string>
 #include <bitmaps/bitmaps_list.h>
 
-#include <wx/string.h> // wxT
 
 struct BITMAP_INFO
 {
-    BITMAPS  id;
-    wxString filename;
-    int      height;
-    wxString theme;
+    BITMAPS     id;
+    std::string filename;
+    int         height;
+    std::string theme;
 };
 
 
-- 
